src/interfaces.ts

```diff
@@ -0,0 +1,9 @@
+import { Observable } from './Observable';
+
+export type UnaryFunction<T, R> = (source: T) => R;
+
+export type OperatorFunction<T, R> = UnaryFunction<Observable<T>, Observable<R>>;
+
+export type FactoryOrValue<T> = T | (() => T);
+
+export type MonoTypeOperatorFunction<T> = OperatorFunction<T, T>;
```

Comment:
nit: we'll likely to have `OperatorFunction<T, R=T>` in `next` if we pick this up into, possible candidate for breaking changes.